### ISSUE TITLE
Problems in data streaming using Brainflow using muse2

### ISSUE THREAD

**Issue by Parvfect** [NeuroTechX/EEG-ExPy#199]
### Discussed in https://github.com/NeuroTechX/eeg-notebooks/discussions/198

<div type='discussions-op-text'>

<sup>Originally posted by **Parvfect** July 26, 2022</sup>
I am trying to run the N170 experiment, and it seemed to be working perfectly before and I have some good data from then. I am not sure what I changed, but the instruction screen seems to come up twice, the second time after running the main experiment for a few seconds.  About two images get displayed when the same screen comes up for the second time. The code used is the following 

```python
from eegnb.devices.eeg import EEG
from eegnb.experiments.visual_n170 import n170

board_name = 'muse2'

record_duration = 120
eeg_device = EEG(device=board_name)
n170.present(duration=record_duration, eeg=eeg_device)
```

And the runtime error is the following, 

<img width="949" alt="issue_1" src="https://user-images.githubusercontent.com/65726543/181122822-11003039-7df2-45f2-afb5-3b5ca1ecdc54.png">
<img width="960" alt="issue_2" src="https://user-images.githubusercontent.com/65726543/181122819-1c37bd44-38b0-4f44-b576-f1fc199fb002.png">

<img width="656" alt="issue_3" src="https://user-images.githubusercontent.com/65726543/181122816-9bf46249-c94c-4920-835b-65f20a640ebd.png">

I am not very clear why this is happening and would really apprectiate some help. I was testing the ExperimentSubClass2 branch when this happened and switched to Master and got the same issue. 
</div>

**Comment by JohnGriffiths** [comment]
Can you repeat and show screen outputs for device = `muse2_bfn` ?

**Comment by Parvfect** [comment]
I get a key-error when I run with muse2_bfn as the device. Am I doing something wrong?

**Comment by JohnGriffiths** [comment]
Yes something not right if you're using the above script and getting a key error.

We need to get back to step 1. 

Please try:

- fresh clone of eeg-notebooks (rename as '_temp' or '_v2' or whatev)
- go in to folder, create new environment, activate environment, `pip install -e .`
- Turn on muse 
- On command line `eegnb checksigqual -ed muse2_bfn`
- If not working: repeat a few times; make sure bluetooth is on
- If you get errors from this, please share screenshot here.

**Comment by Parvfect** [comment]
So it works with muse2 (the signal quality check), but if I do muse2_bfn, I get this
<img width="787" alt="image" src="https://user-images.githubusercontent.com/65726543/181478996-ea80c4de-0a6b-4315-9ce2-cb020f7ffdda.png">

**Comment by JohnGriffiths** [comment]
Do you mean it works with muse2 when bluemuse is running? 

Try one more time: close conda terminal, close bluemuse, open new conda terminal, run _bfn signal check. 

(Brainflow can get confused if there's been other recent communication with the device by currently running programs)

**Comment by Parvfect** [comment]
Yeah so it works with bluemuse and muse2. On closing bluemuse and independently running brainflow via the _bfn run check, I get this error as before
![image](https://user-images.githubusercontent.com/65726543/181496414-25c5f0e5-ec3d-450c-8551-5605ad7f8250.png)

**Comment by JohnGriffiths** [comment]
Next steps on this: 

- try again a few times; reset computer, turn off bluetooth, reset muse, etc. 
- do you have another computer you could try this on by any chance? 
- if problem still persists then post issue in `brainflow-dev/brainflow` github repo with the screenshot you took using the brainflow calls directly. Key information here would be if the `enable_dev_board_logger` has added any usable error info, such as a text snippet with error number or message or something 
- drop link to the brainflow-posted issue in comment on this thread, and also link back to this thread on the brainflow issue. Also repeat the brainflow issue post, at least in part, on this thread for clarity. 

- I believe brainflow has been having some issues with some Windows 11 releases. Solution may be to compile brainflow locally, rather than `pip install`. A bit of a drag but not a huge deal, easy to do. Let's see what Andrey says.

**Comment by Parvfect** [comment]
https://github.com/brainflow-dev/brainflow/issues/531

Issue opened in Brainflow

Unable to prepare the streaming session after finding Muse2 using direct calls to Brainflow.

Code: 

```python
from brainflow.board_shim import BoardShim, BoardIds, BrainFlowInputParams

params = BrainFlowInputParams()
board_id = BoardIds.MUSE_2_BOARD

board = BoardShim(board_id, params)
board.enable_dev_board_logger()

board.prepare_session()
```

Screenshot of Error
![image](https://user-images.githubusercontent.com/65726543/182644268-65004960-2677-4a34-bfdc-da994ebb5cf1.png)

**Comment by Parvfect** [comment]
Original problem with respect to data streaming has been solved by using BlueMuse backend. The problem was that my file had an open call to streaming which was fixed by adding the following, 

```python

if __name__ == '__main__':
    # Do the Experiment

```

However, setting up a streaming session using Brainflow is still unresolved and listed above.

**Comment by oreHGA** [comment]
Summary:
- Suggestion from linked brainflow thread https://github.com/brainflow-dev/brainflow/issues/531 was to update to 3.10
